tests/aead_tests.rs

```diff
@@ -50,7 +50,7 @@ macro_rules! test_aead {
             $(
                 #[allow(non_snake_case)]
                 mod $alg { // Provide a separate namespace for each algorithm's test.
-                    use super::super::{*, test};
```

Comment:
I think you'll have to undo this change? I put this ugliness there because the tests wouldn't pass correctly in wasm32 targets without it.

Reply:
Ok, I'll split it out into its own line with its own `#[allow(unused_imports)]`. But if you'd prefer to just leave the warning in place let me know.

Reply:
I understand now that you're just trying to fix a warning. It turns out that our warning detection in CI was broken. I fixed it in PR #1260. So, you should be able to rebase this on top of the current `main` and you shouldn't need to make changes here any more.